src/javascript/tests/booleans.rs

```diff
@@ -59,3 +59,30 @@ function go() {
 "#
     );
 }
+
+#[test]
+fn matching_booleans() {
+    assert_js!(
+        r#"
+fn go(x, y) {
+    let True = x
+    let False = x
+    let Nil = y
+}
+"#,
+        r#""use strict";
+
+function go(x, y) {
+  let gleam$tmp = x;
+  if (!(gleam$tmp === true)) throw new Error("Bad match");
+  
+  let gleam$tmp = x;
```

Comment:
This is invalid JS, we need to use a different name per assignment otherwise it will crash on attempting to redefine it.

Reply:
Fixed this by introducing a `tmp_var` that is incremented using the next_local_variable_function.
This passes a `Document<'a>` around that get's cloned quite a lot. I think this is correct because it potentially is rendered into the output document multiple times.